src/react-queries/queries/accountManagement.ts

```diff
@@ -14,17 +14,29 @@ const getAccountIdsWithActivityAndMetadata = async (
 ) => {
   const input = { accountNumber, chainIds: [chainId], wallet }
   const accountIdsAndMetadata = await deriveAccountIdsAndMetadata(input)
-  const [[accountId, accountMetadata]] = Object.entries(accountIdsAndMetadata)
 
-  const { account: pubkey } = fromAccountId(accountId)
-  const adapter = assertGetChainAdapter(chainId)
-  const account = await adapter.getAccount(pubkey)
-  const hasActivity = checkAccountHasActivity(account)
+  return Promise.all(
+    Object.entries(accountIdsAndMetadata).map(async ([accountId, accountMetadata]) => {
+      const { account: pubkey } = fromAccountId(accountId)
+      const adapter = assertGetChainAdapter(chainId)
+      const account = await adapter.getAccount(pubkey)
```

Comment:
Do we want to leverage the cached RTK `getAccount` queries here? i.e
```suggestion
      const account = await dispatch(        portfolioApi.endpoints.getAccount.initiate(
          { accountId, upsertOnFetch: true },
          { forceRefetch: true },
        ),
      )
```

This would ensure we do not call `adapter.getAccount(pubkey)` on accounts we already have fetched before. The slight downside to this is there is one side effect on that query (assets upsertion) but no network calls side effects, so I believe we should be good to use it.

Reply:
We'll need to re-evaluate as this is entirely stale with the 3 PR following this one

Reply:
Resolving for now with DM convo started.